free/src/main/scala/cats/free/Cofree.scala

```diff
@@ -0,0 +1,151 @@
+package cats
+package free
+
+/**
+  * A free comonad for some branching functor `S`. Branching is done lazily using Eval.
```

Comment:
This is super nitpicky but could you put `[[]]` around `Eval` and the `Free` in the next line? When I'm reading Scaladocs I always get pissed off if the authors make me search to follow a reference to another class.